Yash/ObserverPattern/src/WeatherStation.cpp

```diff
@@ -0,0 +1,60 @@
+#include <inc/WeatherStation.hpp>
+#include <inc/WeatherObserver.hpp>
+#include <inc/WeatherData.hpp>
+
+////////////////////
+// WeatherStation //
+////////////////////
+WeatherStation::WeatherStation(const std::string& place) : mObservers(new WeatherObserverPtrArray()), mWeatherInfo(), mPlace(place) {}
+
+WeatherStation::~WeatherStation() {}
+
+WeatherStationPtr WeatherStation::create(const std::string& place) {
+	return WeatherStationPtr(new WeatherStation(place));
+}
+
+WeatherObserverPtrArrayPtr WeatherStation::getObservers() {
+	return mObservers;
+}
+
+WeatherDataConstPtr WeatherStation::getWeatherData() {
+	return mWeatherInfo;
+}
+
+std::string WeatherStation::getLocation() const {
+	return mPlace;
+}
+
+void WeatherStation::setNewWeatherData(WeatherDataConstPtr const& newWeatherInfo) {
+	if (newWeatherInfo->getLocation() != mPlace) {
+		throw "This is a weather station for " + mPlace + ". Will accept data only relevant to this location!";
+	}
+	mWeatherInfo = newWeatherInfo;
+	measurementsChanged();

```

Comment:
Can directly call notifyObserver, measurementChanged is being used only once, and it has no other logic than calling notify.

Reply:
Makes sense